Gems/ROS2/Code/Source/Gripper/GripperActionServer.cpp

```diff
@@ -0,0 +1,142 @@
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::EntityId,
+                        &GripperActionServer::m_gripperActionServerName,
+                        "Gripper Action Server",
+                        "Gripper Action Server.");
+            }
+        }
+    }
+
+    bool GripperActionServer::IsGoalActiveState() const
+    {
+        return m_goalHandle->is_active() || m_goalHandle->is_executing() || m_goalHandle->is_canceling();
+    }
+
+    void GripperActionServer::OnTick(float delta, AZ::ScriptTimePoint timePoint)
+    {
+        if (m_goalHandle && IsGoalActiveState())
+        {
+            auto feedback = std::make_shared<GripperCommand::Feedback>();
+            float position = 0.0;
+            float effort = 0.0;
+            bool isDone = false;
+
+            GripperRequestBus::EventResult(position, GetEntityId(), &GripperRequestBus::Events::GetGripperPosition);
+            GripperRequestBus::EventResult(effort, GetEntityId(), &GripperRequestBus::Events::GetGripperEffort);
+            GripperRequestBus::EventResult(isDone, GetEntityId(), &GripperRequestBus::Events::IsGripperReachedGoal);
+
+            if (isDone)
+            {
+                AZ_Printf("GripperActionServer::OnTick", "GripperActionServer::OnTick: Gripper reached goal!");
+                auto result = std::make_shared<GripperCommand::Result>();
+                result->position = position;
+                result->effort = effort;
+                result->reached_goal = true;
+                m_goalHandle->succeed(result);
+            }
+            else
+            {
+                feedback->position = position;
+                feedback->position = effort;
+                feedback->reached_goal = false;
+                m_goalHandle->publish_feedback(feedback);
+            }
+        }
+    }
+
+    rclcpp_action::GoalResponse GripperActionServer::handleGoal(
+        const rclcpp_action::GoalUUID& uuid, std::shared_ptr<const GripperCommand::Goal> goal)
+    {
+        (void)uuid;
+        AZ::Outcome<void, AZStd::string> commandOutcome = AZ::Failure(AZStd::string("No gripper component found!"));
+        GripperRequestBus::EventResult(
+            commandOutcome, GetEntityId(), &GripperRequestBus::Events::GripperCommand, goal->command.position, goal->command.max_effort);
+        if (!commandOutcome.IsSuccess())
+        {
+            AZ_TracePrintf(
+                "GripperActionServer::handleGoal",
+                "GripperActionServer::handleGoal: GripperCommand could not be accepted: %s",
+                commandOutcome.GetError().c_str());
+            return rclcpp_action::GoalResponse::REJECT;
+        }
+        AZ_TracePrintf("GripperActionServer::handleGoal", "GripperActionServer::handleGoal: GripperCommand accepted!");
```

Comment:
`AZ_TracePrintf` is deprecated
```suggestion
        AZ_Trace("GripperActionServer::handleGoal", "GripperActionServer::handleGoal: GripperCommand accepted!\n");
```